x/twap/store.go

```diff
@@ -163,9 +163,10 @@ func (k Keeper) getAllHistoricalPoolIndexedTWAPs(ctx sdk.Context) ([]types.TwapR
 }
 
 // StoreNewRecord stores a record, in both the most recent record store and historical stores.
-func (k Keeper) StoreNewRecord(ctx sdk.Context, twap types.TwapRecord) {
+// NOTE: if it's a new Record make sure to update asset0 and asset1 as well.
+func (k Keeper) StoreNewRecord(ctx sdk.Context, asset0 string, asset1 string, twap types.TwapRecord) {
```

Comment:
note: i had to do this annoying thing where we couldn't just storeNewTwapRecord because the keys are stored using {denom0, denom1} so if we flip to {denom1,denom0} it'd create a newTwapRecord. So i did this so that we can modify existing Key's value with new value. Please lmk if there's a better way of handling this

Reply:
Consider deleting all old records after writing new keys. Might be cleaner than having an API break

Reply:
it's same keys but we update the record

Reply:
What I'm suggesting is:
1. Create duplicate copies where record denoms are switched (and key order is switched)
2. Delete all old records where the key order and assets in the record are switched

The benefit of this approach is avoiding the API break that introduces redundant parameters

Reply:
yup this sounds good, i changed it